include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -0,0 +1,552 @@
+
+        switch (sweep_parameter)
+        {
+            case operational_domain::sweep_parameter::EPSILON_R:
+            {
+                sim_parameters.epsilon_r = val;
+                break;
+            }
+            case operational_domain::sweep_parameter::LAMBDA_TF:
+            {
+                sim_parameters.lambda_tf = val;
+                break;
+            }
+            case operational_domain::sweep_parameter::MU_MINUS:
+            {
+                sim_parameters.mu = val;
+                break;
+            }
+            default:
+            {
+                assert(false && "Unknown sweep parameter");
+            }
+        }
+    }
+    /**
+     * Helper function that sets the value of the x dimension in the simulation parameters.
+     *
+     * @param sim_params Simulation parameter object to set the x dimension value of.
+     * @param val Value to set the x dimension to.
+     */
+    inline void set_x_dimension_value(sidb_simulation_parameters& sim_params, const double val) const noexcept
+    {
+        set_dimension_value(sim_params, val, sweep_dimension::X);
+    }
+    /**
+     * Helper function that sets the value of the y dimension in the simulation parameters.
+     *
+     * @param sim_params Simulation parameter object to set the y dimension value of.
+     * @param val Value to set the y dimension to.
+     */
+    inline void set_y_dimension_value(sidb_simulation_parameters& sim_params, const double val) const noexcept
+    {
+        set_dimension_value(sim_params, val, sweep_dimension::Y);
+    }
+    /**
+     * Performs an exhaustive ground state simulation for all input combinations of the stored layout using the given
+     * simulation parameters. This function is used by all operational domain computation techniques. The function
+     * terminates as soon as a non-operational state is found. In the worst case, the function performs \f$ 2^n \f$
+     * simulations, where \f$ n \f$ is the number of inputs of the layout.
+     *
+     * @param sim_params Simulation parameters to use for the simulation.
+     * @return The operational status of the layout under the given simulation parameters.
+     */
+    [[nodiscard]] operational_domain::operational_status
+    is_operational(const sidb_simulation_parameters& sim_params) noexcept
+    {
+        // increment the number of evaluated parameter combinations
+        ++stats.num_evaluated_parameter_combinations;
+
+        // take the first (and only) output BDL pair
+        const auto& output_bdl_pair = output_bdl_pairs.front();
+
+        // reset the BDL input iterator
+        bii = 0ull;
+
+        // for each input combination
+        for (auto i = 0u; i < truth_table.num_bits(); ++i, ++bii)
+        {
+            // the expected output of the layout is the i-th bit of the truth table
+            const auto expected_output = kitty::get_bit(truth_table, i);
+
```

Comment:
As discussed, I think we need a check here like ``is_three_state_simulation_required(*bii)``. After that the question is whether the simulation should be continued with 3-state simulation or whether the layout should be marked as ``NON_OPERATIONAL``. I tend to the latter.

Reply:
Is this function available outside of your QuickExact branch already?

Reply:
Not yet, but I will generate a PR today.

Reply:
Many thanks!